src/fs/metadata_log.cc

```diff
@@ -89,6 +94,11 @@ future<> metadata_log::shutdown() {
     });
 }
 
+void metadata_log::switch_fs_read_only_mode(bool val) noexcept {
```

Comment:
We have a nice idiom for naming bool parameters - `bool class` from `seastar/util/bool_class.hh` - please consider using it - it provides more context and type safety.